cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -564,4 +568,125 @@ public InputStream getInputStream() throws IOException {
+     */
+    @BeforeClass
+    public static void reduceBackoffTime() throws Exception {
+        // Reflectively reduce the time limit for testing
+        Field f = RequestLimitInterceptor.class.getDeclaredField("initialSleep");
+        f.setAccessible(true);
+        f.set(null, 1); // Set the value to 1 ms
+    }
+
+    /**
+     * Test that a request is replayed in response to a 429.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void test429Backoff() throws Exception {
+        mockWebServer.enqueue(response429);
+        mockWebServer.enqueue(new MockResponse());
+
+        CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                .build();
+        c.executeRequest(Http.GET(c.getBaseUri()));
+
+        assertEquals("There should be 2 requests", 2, mockWebServer.getRequestCount());
+    }
+
+    /**
+     * Test that the maximum number of retries is reached and the backoff is of at least the
+     * expected duration.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void test429BackoffMaxDefault() throws Exception {
+
+        // Always respond 429 for this test
+        mockWebServer.setDispatcher(new Dispatcher() {
+            @Override
+            public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+                return response429;
+            }
+        });
+
+
+        long startTime = System.currentTimeMillis();
+        try {
+            CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                    .build();
+            c.executeRequest(Http.GET(c.getBaseUri()));
+        } catch (TooManyRequestsException e) {
+            long duration = System.currentTimeMillis() - startTime;
+            // 9 backoff periods for 10 attempts: 1 + 2 + 4 + 8 + 16 + 32 + 64 + 128 + 256 = 511 ms
+            assertTrue("The duration should be at least 511 ms, but was " + duration, duration >=
+                    511);
+            assertEquals("There should be 10 request attempts", 10, mockWebServer
+                    .getRequestCount());
+        }
+
+    }
+
+    /**
+     * Test that the number of retries is configurable.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void test429BackoffMaxConfigured() throws Exception {
+
+        // Always respond 429 for this test
+        mockWebServer.setDispatcher(new Dispatcher() {
+            @Override
+            public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
+                return response429;
+            }
+        });
+
+
+        long startTime = System.currentTimeMillis();
+        try {
+            CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                    .build();
+            c.executeRequest(Http.GET(c.getBaseUri()).setNumberOfRetries(3));
```

Comment:
The setter for retries has always exposed on `HttpConnection`, I added a getter so that the interceptor could use the value.
I think 10 is a sensible default and if someone really wants to configure it they can use a custom `HttpConnection` and `executeRequest`.